### PR TITLE
feat: Add cookie consent to YouTubeVideoModal

### PR DESCRIPTION
**Pull request by KarlEST**
## 🖼 Context

We need to make sure that everybody who uses `YouTubeVideoModal` will respect the cookie consent.

## 🚀 Changes

YouTube video opens up in a new tab if you don't have cookie consent

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/marketing-components/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary
- [x] Change has been approved by someone outside of your team

### CODE REVIEW

File: packages/marketing-components/src/youtubevideomodal/YouTubeVideoModal.js [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
   const posterFallbackUrl = `https://img.youtube.com/vi/${videoId}/maxresdefault.jpg`;
   const videoUrl = `https://www.youtube-nocookie.com/embed/${videoId}?enablejsapi=1&html5=1&rel=0&showinfo=0&autoplay=1&wmode=opaque`;
+  let isVideoModalOpen = false;
+
+  if (isOpen) {
+    if (hasValidConsent()) {
+      isVideoModalOpen = true;
+    } else if (typeof window !== 'undefined') {
+      window.open(`https://youtu.be/${videoId}`, '_blank');
+      onDismiss();
```
Comment (karanjthakkar) [inline packages/marketing-components/src/youtubevideomodal/YouTubeVideoModal.js:17]: Should there be a `return null` in this block so if there is no consent you don't end up rendering the video modal component anyway?

Comment (KarlEST) [reply packages/marketing-components/src/youtubevideomodal/YouTubeVideoModal.js:17]: Hmm, currently this block relies on `isOpen` to be `true` which it isn't in the beginning(only after button click).
If we don't want to render anything unless we have a consent then I think it makes sense to move this logic out of this block and check for consent separately

Comment (KarlEST) [reply packages/marketing-components/src/youtubevideomodal/YouTubeVideoModal.js:14]: Updated so now we don't render `VideoModal` if we don't have a consent

**Review by danlaush — APPROVED** [review]
Lookin good to me 👍

**Pull request merged**